### PR TITLE
Add documentation for build's `--type` flag

### PR DESCRIPTION
**Pull request by robbertvanginkel**
Added in https://github.com/bufbuild/buf/pull/949, released in https://github.com/bufbuild/buf/releases/tag/v1.1.0.

### CODE REVIEW

File: docs/build/usage.md [hunk 1]
```diff
@@ -283,6 +283,75 @@ command:
 $ buf build --path path/to/foo.proto --path path/to/bar.proto
 ```
 
+## Limit to specific types
+
+The output FileDescritptorSet or image from a `buf build` will contain all types
```
Comment (doriable) [inline docs/build/usage.md:288]: `%s/FileDescritptorSet/FileDescriptorSet`. As a side note, I think we show `FileDescriptorSet` with the code backticks elsewhere in the documentation, so we may want to do that here also.

File: docs/build/usage.md [hunk 2]
```diff
@@ -283,6 +283,75 @@ command:
...
+declared in the module. For advanced usecases, users may want an image or
```
Comment (doriable) [inline docs/build/usage.md:289]: `%s/usecases/use cases`

File: docs/build/usage.md [hunk 3]
```diff
@@ -283,6 +283,75 @@ command:
...
+FileDescriptorSet containing only a subset of types. Introduced in `v1.1`, the
```
Comment (doriable) [inline docs/build/usage.md:290]: `%s/v1.1/v1.1.0`? i think this could go either way, tbh.

File: docs/build/usage.md [hunk 4]
```diff
@@ -283,6 +283,75 @@ command:
...
+`--type` flag accepts fully qualified protobuf names and limits the output image
```
Comment (doriable) [inline docs/build/usage.md:291]: I believe "Protobuf" is capitalised in other parts of our documentation.

File: docs/build/usage.md [hunk 5]
```diff
@@ -283,6 +283,75 @@ command:
...
+to contain only descriptors required to represent those types and their required
+dependencies. The type flag supports messages, enums and services.
```
Comment (doriable) [inline docs/build/usage.md:293]: Not to be "that guy" but there's an oxford comma later on in the docs, so we should probably use one here also.

File: docs/build/usage.md [hunk 6]
```diff
@@ -283,6 +283,75 @@ command:
...
+    - the parent message if this message is a nested definition
+    - Custom options used in the enum, enum values, and the file
+      in which the message is defined
+  - Services
+    - request & response types referenced in methods
+    - custom options for the service, its methods, and the file
+      in which the message is defined
+
+As an example, consider the these sample protos:
+```proto
```
Comment (doriable) [inline docs/build/usage.md:318]: Does `proto` work? Looking at the other files in the docs, it seems that `protobuf` is the language setting for the code block.

I like the example though, makes the behaviour very clear :)

Comment (robbertvanginkel) [reply docs/build/usage.md]: Proto did work for me locally, but I've updated to `protobuf` to align with the others.

File: docs/build/usage.md [hunk 7]
```diff
@@ -283,6 +283,75 @@ command:
 $ buf build --path path/to/foo.proto --path path/to/bar.proto
 ```
 
+## Limit to specific types
+
+The output `FileDescriptorSet` or image from a `buf build` will contain all types
+declared in the module. For advanced use cases, users may want an image or
+FileDescriptorSet containing only a subset of types. Introduced in `v1.1.0`, the
```
Comment (doriable) [inline docs/build/usage.md:290]: Backticks for `FileDescriptorSet` ^^"

**Review by doriable — APPROVED** [review]
Tiny nit, but stamping early, as I agree with the content and don't want to block. @lucperkins should have the final say though :)

**Comment by lucperkins** [comment]
I have some desired tech-writer-ly changes but I can push that through in a follow-up PR. LGTM.

**Review by lucperkins — APPROVED** [review]

File: docs/build/usage.md (hunk 7 above)
Comment (robbertvanginkel) [inline docs/build/usage.md:290]:
```suggestion
`FileDescriptorSet` containing only a subset of types. Introduced in `v1.1.0`, the
```

**Pull request merged**